packages/atclient/src/metadata.c

```diff
@@ -684,70 +684,70 @@ int atclient_atkey_metadata_to_protocolstr(const atclient_atkey_metadata metadat
   atclient_atstr_init(&protocolstr, 4096);
 
   if (atclient_atkey_metadata_is_ttl_initialized(metadata)) {
-    if ((ret = atclient_atstr_append(&protocolstr, ":ttl:%ld", metadata.ttl) != 0)) {
+    if ((ret = atclient_atstr_append(&protocolstr, "ttl:%ld:", metadata.ttl) != 0)) {
```

Comment:
prepended `:` is preferred here, a trailing `:` could result in an invalid protocol command if that part is the final segment in the command.

Reply:
You make a good point

I reverted the change in https://github.com/atsign-foundation/at_c/pull/156/commits/cb185fe4e01d6d6721f8117ae3187a19e93f5d90